src/paint/map_element/fence.c

```diff
@@ -14,14 +14,373 @@
  *****************************************************************************/
 #pragma endregion
 
-#include "map_element.h"
+#include "../../common.h"
 #include "../../addresses.h"
 #include "../../world/map.h"
+#include "../../drawing/drawing.h"
+#include "../../world/scenery.h"
+#include "../../game.h"
+#include "../../ride/track.h"
+#include "../../config.h"
+#include "../../localisation/localisation.h"
+#include "../../interface/colour.h"
+#include "../../interface/viewport.h"
+#include "../paint.h"
 
-extern bool TempForScrollText;
+const uint8 byte_9A406C[] = {
+    2, 2, 22, 26, 30, 34, 34, 34, 34, 34, 30, 26, 22, 2, 6, 2,
+    2, 2, 6, 10, 14, 18, 18, 18, 18, 18, 14, 10, 6, 2, 22, 2
+};
 
-void fence_paint(uint8 direction, int height, rct_map_element* mapElement) {
-	TempForScrollText = true;
-	RCT2_CALLPROC_X(0x6E44B0, 0, 0, direction, height, (int)mapElement, 0, 0);
-	TempForScrollText = false;
+const uint8 byte_9A408C[] = {
+    0, 0, 4, 8, 12, 16, 16, 16, 16, 16, 12, 8, 4, 0, 20, 0,
+    0, 0, 20, 24, 28, 32, 32, 32, 32, 32, 28, 24, 20, 0, 4, 0
+};
+
+const uint8 byte_9A40AC[] = {
+    2, 2, 6, 10, 14, 18, 18, 18, 18, 18, 14, 10, 6, 2, 22, 2,
+    2, 2, 22, 26, 30, 34, 34, 34, 34, 34, 30, 26, 22, 2, 6, 2
+};
+const uint8 byte_9A40CC[] = {
+    0, 0, 20, 24, 28, 32, 32, 32, 32, 32, 28, 24, 20, 0, 4, 0,
+    0, 0, 4, 8, 12, 16, 16, 16, 16, 16, 12, 8, 4, 0, 20, 0
+};
+
+void fence_paint_door(uint32 imageId,
+                      rct_scenery_entry * sceneryEntry,
+                      uint32 imageColourFlags, uint32 tertiaryColour, uint32 dword_141F710,
+                      rct_xyz16 offset,
```

Comment:
offset should be converted or defined as a rct_sxyz8 to get rid of the warnings.